drivers: sensor: Move to vendor subdirectory

MaureenHelm opened the pull request:
Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: adi: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: bosch: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: st: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: maxim: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: microchip: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: nxp: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: silabs: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: ti: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: ams: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: sensirion: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: wsen: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: ite: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: nordic: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: nuvoton: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: rohm: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: seeed: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

teburd commented:
very nice

avisconti commented:
Not a bad idea and I personally like it.

teburd commented on drivers/sensor/CMakeLists.txt:
```diff
@@ -1,179 +1,73 @@
-add_subdirectory_ifdef(CONFIG_TSL2561 tsl2561)
-add_subdirectory_ifdef(CONFIG_TSL2591 tsl2591)
-add_subdirectory_ifdef(CONFIG_VCMP_IT8XXX2 ite_vcmp_it8xxx2)
 add_subdirectory_ifdef(CONFIG_VCNL4040 vcnl4040)
```
these three are vishay and could probably be moved as well correct?

teburd commented on drivers/sensor/Kconfig:
```diff
@@ -81,179 +81,74 @@ config SENSOR_INFO
-source "drivers/sensor/qdec_mcux/Kconfig"
-source "drivers/sensor/qdec_nxp_s32/Kconfig"
-source "drivers/sensor/qdec_nrfx/Kconfig"
 source "drivers/sensor/qdec_sam/Kconfig"
```
microchip

teburd commented on drivers/sensor/CMakeLists.txt:
```diff
@@ -1,179 +1,73 @@
-add_subdirectory_ifdef(CONFIG_QDEC_MCUX qdec_mcux)
-add_subdirectory_ifdef(CONFIG_QDEC_NXP_S32 qdec_nxp_s32)
-add_subdirectory_ifdef(CONFIG_QDEC_NRFX qdec_nrfx)
 add_subdirectory_ifdef(CONFIG_QDEC_SAM qdec_sam)
```
microchip

teburd reviewed (dismissed):
Minor nits of some stuff that can be moved later or now

avisconti reviewed (approved):
LGTM
Reviewed the commit for ST part more carefully than the others

avisconti reviewed (changes requested):
Sorry, I just noticed that two recent drivers are missing in ST subdir:

- iis328dq
- lis2dux12

Rest is fine. Thx!

MaureenHelm committed:
drivers: sensor: vishay: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: aosong: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: honeywell: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: infineon: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: espressif: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: asahi_kasei: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

MaureenHelm committed:
drivers: sensor: meas: Move to vendor subdirectory

Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>
